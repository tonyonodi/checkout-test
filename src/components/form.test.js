import React from "react";
import renderer from "react-test-renderer";
import Form from "./form";

it("renders inputs correctly", () => {
  const tree = renderer
    .create(
      <Form
        values={{
          name: "Alice",
          email: "alice@gmail.com",
          rating: 3,
          comment: "Quick comment",
        }}
        errors={{}}
        updateValues={() => {}}
        addComment={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders errors correctly", () => {
  const tree = renderer
    .create(
      <Form
        values={{
          name: "Alice",
          email: "alice@gmail.com",
          rating: 3,
          comment: "Quick comment",
        }}
        errors={{
          name: "Name cannot be empty.",
          email: "Email cannot be empty.",
          rating: "Name cannot be empty.",
        }}
        updateValues={() => {}}
        addComment={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
