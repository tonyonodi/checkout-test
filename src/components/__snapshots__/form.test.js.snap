// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders errors correctly 1`] = `
<form
  className="sc-bdVaJa bPGLpq"
  onSubmit={[Function]}
>
  <div>
    <label
      className="sc-bwzfXH uTxCW"
    >
      Name *
    </label>
    <div
      className="sc-ifAKCX ifRUKc"
    >
      Name cannot be empty.
    </div>
    <input
      className="sc-htpNat drnJZU"
      name="name"
      onChange={[Function]}
      type="text"
      value="Alice"
    />
  </div>
  <div>
    <label
      className="sc-bwzfXH uTxCW"
    >
      Email *
    </label>
    <div
      className="sc-ifAKCX ifRUKc"
    >
      Email cannot be empty.
    </div>
    <input
      className="sc-htpNat drnJZU"
      name="email"
      onChange={[Function]}
      type="email"
      value="alice@gmail.com"
    />
  </div>
  <div>
    <label
      className="sc-bwzfXH uTxCW"
    >
      Rating *
    </label>
    <div
      className="sc-ifAKCX ifRUKc"
    >
      Name cannot be empty.
    </div>
    <input
      className="sc-htpNat drnJZU"
      max="5"
      min="0"
      name="rating"
      onChange={[Function]}
      step="1"
      type="number"
      value={3}
    />
  </div>
  <div />
  <label
    className="sc-bwzfXH uTxCW"
  >
    Comment
  </label>
  <textarea
    className="sc-bxivhb dqNOXg"
    name="comment"
    onChange={[Function]}
    value="Quick comment"
  />
  <input
    className="sc-htpNat drnJZU"
    type="submit"
  />
</form>
`;

exports[`renders inputs correctly 1`] = `
<form
  className="sc-bdVaJa bPGLpq"
  onSubmit={[Function]}
>
  <div>
    <label
      className="sc-bwzfXH uTxCW"
    >
      Name *
    </label>
    <input
      className="sc-htpNat drnJZU"
      name="name"
      onChange={[Function]}
      type="text"
      value="Alice"
    />
  </div>
  <div>
    <label
      className="sc-bwzfXH uTxCW"
    >
      Email *
    </label>
    <input
      className="sc-htpNat drnJZU"
      name="email"
      onChange={[Function]}
      type="email"
      value="alice@gmail.com"
    />
  </div>
  <div>
    <label
      className="sc-bwzfXH uTxCW"
    >
      Rating *
    </label>
    <input
      className="sc-htpNat drnJZU"
      max="5"
      min="0"
      name="rating"
      onChange={[Function]}
      step="1"
      type="number"
      value={3}
    />
  </div>
  <div />
  <label
    className="sc-bwzfXH uTxCW"
  >
    Comment
  </label>
  <textarea
    className="sc-bxivhb dqNOXg"
    name="comment"
    onChange={[Function]}
    value="Quick comment"
  />
  <input
    className="sc-htpNat drnJZU"
    type="submit"
  />
</form>
`;
